### ISSUE TITLE
[GR-44523] Implement Tail Call Proposal for WebAssembly

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#9763]
### TL;DR

We plan to implement [the Tail Call proposal](https://github.com/WebAssembly/tail-call) in GraalWasm.

### Details

This proposal adds new call instructions for tail calls. Tail calls behave like regular function calls, except that they pop the current stack frame before executing the call. Support for such calls is necessary for a correct and efficient implementation of languages that rely on tail call elimination. It also facilitates implementing languages with control constructs like coroutines or continuations efficiently.
Implementing this proposal will allow us to run such languages efficiently on GraalWasm.

More details at https://github.com/WebAssembly/tail-call/blob/main/proposals/tail-call/Overview.md
